qiskit/pulse/schedule.py

```diff
@@ -724,34 +852,69 @@ def __call__(self, *args: List[Union[float, complex]],
         return self.bind_parameters(*args, **kwargs)
 
 
-def _insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
+def _find_possible_interval_index(intervals: List[Interval],
+                                  interval: Interval,
+                                  index: int = 0) -> int:
+    """Using binary search on start times, find an interval.
+    Args:
+        intervals: A sorted list of non-overlapping Intervals.
+        interval: The interval for which the index into intervals will be found.
+        index: A running tally of the index, for recursion. The user should not pass a value.
+    Returns:
+        The index into intervals that new_interval would be inserted to maintain
+        a sorted list of intervals.
+    """
+    if not intervals or len(intervals) == 1:
+        return index
+
+    mid_idx = len(intervals) // 2
+    mid = intervals[mid_idx]
+    if interval[1] <= mid[0] and (interval != mid):
+        return _find_possible_interval_index(intervals[:mid_idx], interval, index=index)
+    else:
+        return _find_possible_interval_index(intervals[mid_idx:], interval, index=index + mid_idx)
+
+
+def _find_interval_index(intervals: List[Interval], interval: Interval) -> int:
```

Comment:
```suggestion
def _index(intervals: List[Interval], interval: Interval) -> int:
```
or
```suggestion
def _index_of(intervals: List[Interval], interval: Interval) -> int:
```
This has the same behavior as list .index

Reply:
Fixed in 0e52dbf15.